dat/missions/dvaered/colonel-transport.lua

```diff
@@ -0,0 +1,163 @@
+--[[
+<?xml version='1.0' encoding='utf8'?>
+<mission name="Dvaered Colonel Escort">
+ <unique />
+ <priority>4</priority>
+ <chance>43</chance>
+ <location>Bar</location>
+ <faction>Dvaered</faction>
+ <cond>require("misn_test").reweight_active()</cond>
+</mission>
+--]]
+--[[
+
+   Mission: Escort a Dvaered colonel
+   Description: Small mission where you escort a Dvaered Arsenal.
+                This is a one-off that's not part of any major
+                storyline; the in-game purpose will remain a mystery.
+
+--]]
+
+local escort = require "escort"
+local fmt = require "format"
+local neu = require "neutral"
+local vn = require "vn"
+local vni = require "vnimage"
+
+local reward = 0
+
+local source_system = nil
+
+lmisn.getSpobAtDistance(sys=Dvaer, 1, 6, fct=faction.get("Dvaered")[, samefct=true[, filter=nil[, data=nil[, hidden=false]]]])
```

Comment:
should probably be:
```lua
local spbs = lmisn.getSpobAtDistance( nil, 4, 6, "Dvaered" )
if #spbs <= 0 then misn.finish(false) end

mem.destspb = spbs[ rnd.rnd(1,#spbs) ]
```

Also, you should put it in the `create()` function block at the top.

Reply:
This will get all spobs that meet the criteria, fail if there are none, and choose a random one to put into `mem.destspb`, that you can use as the destination later.